src/change.c

```diff
@@ -560,6 +560,12 @@ changed_common(
 	    if (!redraw_not_allowed && wp->w_redr_type < UPD_VALID)
 		wp->w_redr_type = UPD_VALID;
 
+	    // Reset "w_skipcol" if the topline length is now smaller.
+	    if (wp->w_topline >= lnum && wp->w_topline < lnume
```

Comment:
I think this save some calls
```suggestion
	    if (wp->w_skipcol > 0
	    	    && wp->w_topline >= lnum && wp->w_topline < lnume
```